### PR TITLE
TypeScript Compiler Testing in Hephaestus

### PR DESCRIPTION
**Pull request by alexisthedev**
This pull request consists of the required modules and additions to test the TypeScript compiler with Hephaestus.

Important additions include:

- AST-to-TypeScript Translator
- Core API extension to allow generation of programs with native types of any language (eg. TypeScript's unique union types)
- Null and Undefined Types
- Null Constant
- Literal Types (Native TypeScript)
- Type Aliases (Native TypeScript)
- Union Types (Native TypeScript)

[A more detailed overview is available here.](https://github.com/alexisthedev/hephaestus/blob/main/GSOC.md)

### CODE REVIEW

**Comment by StefanosChaliasos** [comment]
@alexisthedev, please resolve any conflicts.

File: GSOC.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+
```
Comment (StefanosChaliasos) [inline GSOC.md:1]: Do we need this file in the main repo @theosotr?

File: deployment/run.sh [hunk 2]
```diff
@@ -47,6 +47,13 @@ run_groovy_from_source() {
         --language groovy --cast-numbers
 }
 
+run_typescript() {
+    npm install -g typescript@next
```
Comment (StefanosChaliasos) [inline deployment/run.sh:51]: Typescript should be installed in the setup script. If you want to reinstall the nightly build, then wrap that functionality in an install_nightly function.

**Comment by StefanosChaliasos** [comment]
@alexisthedev, could you please add some example instructions and expected results in the description of that PR?

File: GSOC.md (hunk 1 above)
Comment (alexisthedev) [reply GSOC.md:1]: I forgot it in there. Sorry! Removing it now.

**Pull request closed without merging**